### DISCUSSION TITLE
Docker install Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by factor200x** [netbox-community/netbox-docker#881] [Getting Help]
Hello,

I have installed a VM with Ubuntu 22.04 under Proxmox and have done the installation of Netplan (Docker) as described here (https://github.com/netbox-community/netbox-docker/wiki/Getting-Started). 

When starting via docker-compose up I get the following output: 

`Creating network "netbox-docker_default" with the default driver
Creating volume "netbox-docker_netbox-media-files" with local driver
Creating volume "netbox-docker_netbox-postgres-data" with local driver
Creating volume "netbox-docker_netbox-redis-data" with local driver
Creating volume "netbox-docker_netbox-redis-cache-data" with local driver
Creating netbox-docker_redis-cache_1         ... done
Creating netbox-docker_redis_1       ... done
Creating netbox-docker_postgres_1    ... done
Creating netbox-docker_netbox-housekeeping_1 ...
Creating netbox-docker_netbox-worker_1       ... error

Creating netbox-docker_netbox-housekeeping_1 ... error                                                                                                                             tem

ERROR: for netbox-docker_netbox-housekeeping_1  Cannot start service netbox-housekeeping: error while creating mount source path '/inst/netbox-docker/configuration': mkdir /inst: read-only file system

ERROR: for netbox-worker  Cannot start service netbox-worker: error while creating mount source path '/inst/netbox-docker/scripts': mkdir /inst: read-only file system

ERROR: for netbox-housekeeping  Cannot start service netbox-housekeeping: error while creating mount source path '/inst/netbox-docker/configuration': mkdir /inst: read-only file system
ERROR: Encountered errors while bringing up the project.`

A search on the net for this problem was unfortunately unsuccessful. I would be very grateful if someone could give me a hint what I did wrong.

Thanks
Olaf

**Comment by Foozer** [comment]
Problem is with permissions. You can use _sudo docker-compose up_

**Reply by CaptainBalou** [reply]
I just installed a new **Ubuntu 22.04 LTS minimal** virtual machine on VirtualBox and I have the same issue.
The behavior persists with the use of sudo.
Because I'm very new to docker and also Netbox I have no idea how to solve this. Is it possible to provide more logs somehow?

![image](https://user-images.githubusercontent.com/17015872/201311610-d3ab7bb6-b256-41d2-a6ca-838c973731c6.png)

**Reply by CaptainBalou** [reply]
Looks like this is something regarding Apparmor which is restricting files and directories.
Because my system is just a demo VM I deinstalled it like mentioned in this [article](https://forums.docker.com/t/can-not-stop-docker-container-permission-denied-error/41142/5). I can't recommend this for productive systems.

**Comment by timrabl** [comment]
Hi looks more like a filesystem error to me. The error message is actually quite clear.  
The source path for the Docker bind mount cannot be created because the filesystem is mounted read-only.

Ubuntu 22.04 mounts the root disk read only when filesystem errors occur. The behavior can be found in `/etc/fstab` or by using:

```bash
root@redacted:~# findmnt /
TARGET SOURCE    FSTYPE OPTIONS
/      /dev/vda1 ext4   rw,relatime,discard,errors=remount-ro
```

See [**man mount(8)**](https://linux.die.net/man/8/mount) 
```
errors={continue|remount-ro|panic}
  Define the behaviour when an error is encountered. (Either ignore errors and just mark the 
  filesystem erroneous and continue, or remount the filesystem read-only, or panic and halt the 
  system.) The default is set in the filesystem superblock, and can be changed using [tune2fs]. 
  (https://linux.die.net/man/8/tune2fs)(8).
```

But maybe the read-only mounting has other reasons.  
Without further logs, I think it is difficult to make a clear statement.
